frontend/src/app/admin/Organization/TenantDataAccessForm.test.tsx

```diff
@@ -0,0 +1,69 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { MemoryRouter } from "react-router";
+
+import TenantDataAccessForm from "./TenantDataAccessForm";
+import { OrganizationOption } from "./OrganizationDropDown";
+
+let saveTenantDataAccess: jest.Mock;
+
+const organizations: OrganizationOption[] = [
+  { name: "Org 1 Name", externalId: "ORG_1" },
+  { name: "Org 2 Name", externalId: "ORG_2" },
+];
+
+describe("TenantDataAccessForm", () => {
+  beforeEach(() => {
+    saveTenantDataAccess = jest.fn();
+  });
+
+  it("Submits a valid access form", async () => {
+    render(
+      <MemoryRouter>
+        <TenantDataAccessForm
+          organizationOptions={organizations}
+          organizationExternalId=""
+          justification=""
+          saveTenantDataAccess={saveTenantDataAccess}
+        />
+      </MemoryRouter>
+    );
+    const organizationDropdown = screen.getByTestId("organization-dropdown");
+    fireEvent.change(organizationDropdown, {
+      target: { selectedValue: organizations[0].externalId },
+    });
+    fireEvent.change(organizationDropdown, {
+      target: { value: organizations[0].externalId },
+    });
+    await waitFor(async () => {
+      fireEvent.blur(organizationDropdown);
+    });
+    fireEvent.change(screen.getByLabelText("Justification", { exact: false }), {
+      target: { value: "sample justification text" },
+    });
+    const saveButton = await screen.getAllByText("Save Changes")[0];
+    expect(saveButton).toBeEnabled();
+    await waitFor(async () => {
+      fireEvent.click(saveButton);
+    });
+    expect(saveTenantDataAccess).toBeCalledTimes(1);
```

Comment:
can you check for the new org name to appear in the page after you've granted yourself access? or is that org name not visible on the admin panel?

Reply:
It would be visible, but I don't think checking it here would really test what we want to test.  After tenant data access is changed, `window.location.reload()` is called and the header is populated with the whoami response that will contain the org that was just chosen.